sdk/data-generators/image-processing-generator/src/main.rs

```diff
@@ -29,82 +28,82 @@ fn main() -> Result<(), Box<dyn Error>> {
+                .short("w")
+                .long("width")
+                .value_name("NUMBER")
+                .help("The width of the image to generate")
+                .takes_value(true)
+                .validator(is_u64)
+                .default_value("10"),
+        )
+        .arg(
+            Arg::with_name("height")
+                .short("h")
+                .long("height")
+                .value_name("NUMBER")
+                .help("The height of the image to generate")
+                .takes_value(true)
+                .validator(is_u64)
+                .default_value("10"),
         )
         .get_matches();
 
     let file_prefix = matches
         .value_of("file_prefix")
         .ok_or("Failed to read the file prefix.")?;
+    let image_path = matches
+        .value_of("image_path")
+        .ok_or("Failed to read the image path.")?;
     let image_width = matches
         .value_of("width")
-        .ok_or("Failed to read the width")?;
+        .ok_or("Failed to read the width")?
+        .parse::<u64>()
+        .map_err(|_| "Failed to parse the width.")? as u32;
     let image_height = matches
         .value_of("height")
-        .ok_or("Failed to read the height")?;
+        .ok_or("Failed to read the height")?
+        .parse::<u64>()
+        .map_err(|_| "Failed to parse the height.")? as u32;
 
-    // Download random image
-    let mut downloader = Downloader::builder()
-        .download_folder(std::path::Path::new("./"))
-        .parallel_requests(1)
-        .build()
-        .expect("Failed to configure the downloader");
-    let file_path = format!("{}.dat", file_prefix);
-    let dl =
-        Download::new(format!("https://picsum.photos/{}/{}", image_width, image_height).as_str())
-            .file_name(Path::new(&file_path));
-    let _result = downloader
-        .download(&[dl])
-        .map_err(|e| format!("Failed to download image: {}", e));
+    let output_filename = format!("{}.dat", file_prefix);
 
-    // Convert image to PNG.
-    // This is required by the example, as `image` uses threads to load JPG images, which are not supported in WebAssembly
+    // Read PNG image
     let mut reader =
-        Reader::open(&file_path).map_err(|e| format!("Failed to open image: {}", e))?;
+        Reader::open(&image_path).map_err(|e| format!("Failed to open image: {}", e))?;
     reader.set_format(ImageFormat::Jpeg);
     let img = reader
         .decode()
         .map_err(|e| format!("Failed to load image: {}", e))?;
+
+    // Resize image. Aspect ratio is not preserved
+    let img = img.resize_exact(image_width, image_height, FilterType::Gaussian);
+
+    // Save image as PNG.
+    // This is required by the example, as `image` uses threads to load JPG images,
+    // which are not supported in WebAssembly
     let _result = img
-        .save_with_format(file_path, ImageFormat::Png)
+        .save_with_format(format!("{}", output_filename), ImageFormat::Png)
```

Comment:
Why do you need `format!("{}", output_filename)` here?

Reply:
Right. I guess I initially meant to concatenate the filename with an extension